src/game/enemy/events.rs

```diff
@@ -1,4 +1,7 @@
 use bevy::prelude::Event;
 
 #[derive(Event)]
-pub struct EnemyShotEvent(pub usize);
\ No newline at end of file
+pub struct EnemyShotEvent(pub u32);
+
+#[derive(Event)]
+pub struct EnemyShotPlayerEvent(pub u32); // Do not need param if we don't care which enemy shot
```

Comment:
want to say it would make most sense to leave all the determination on of the bullet makes it to the player on the enemy side, so probably dont need the u32 here. It may even make the most sense to have this event as a "PlayerShotEvent" in the player plugin